File: src/main/java/xyz/upperlevel/hgame/DefaultFont.java
package xyz.upperlevel.hgame;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.freetype.FreeTypeFontGenerator;


public final class DefaultFont {
    public static final BitmapFont FONT;

    static {
        FreeTypeFontGenerator generator = new FreeTypeFontGenerator(Gdx.files.internal("Lato.ttf"));
        FreeTypeFontGenerator.FreeTypeFontParameter parameter = new FreeTypeFontGenerator.FreeTypeFontParameter();
        parameter.size = 32;
        parameter.borderWidth = 2;
        parameter.borderColor = Color.BLACK;

        FONT = generator.generateFont(parameter);
    }

    private DefaultFont() {}
}

File: src/main/java/xyz/upperlevel/hgame/DefaultFont.kt
package xyz.upperlevel.hgame

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.graphics.g2d.freetype.FreeTypeFontGenerator


object DefaultFont {
    val FONT: BitmapFont

    init {
        val generator = FreeTypeFontGenerator(Gdx.files.internal("font.ttf"))
        val parameter = FreeTypeFontGenerator.FreeTypeFontParameter()
        parameter.size = 32
        parameter.borderWidth = 3f
        parameter.borderColor = Color.BLACK

        FONT = generator.generateFont(parameter)
    }
}
